blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -128,13 +166,37 @@ protected MetricData getRollupByGranularity(
         return metricData;
     }
 
+    private MetricData getMetricData(long from, long to, Granularity g, Locator locator) {
+
+        String unit = null;
+        Callable<String> unitsGetter = new UnitsGetter(locator.getTenantId(),locator.getMetricName());
+        Future<String> unitsGetterFuture = executorService.submit(unitsGetter);
+
+        MetricData metricData;
+        metricData = AstyanaxReader.getInstance().getDatapointsForRange(
+                locator,
+                new Range(g.snapMillis(from), to),
+                g);
+
+        try {
+            unit = unitsGetterFuture.get();
+        } catch (InterruptedException ie) {
```

Comment:
If there are problems while getting units, out, should be set it to `UNKNOWN` instead of null or may be just not return unit? What do you think?